packager/react-packager/src/AssetServer/index.js

```diff
@@ -140,6 +140,7 @@ class AssetServer {
         const dir = res[0];
         const files = res[1];
         const assetData = getAssetDataFromName(filename, new Set([platform]));
+        assetData.assetName = decodeURIComponent(assetData.assetName);
```

Comment:
Consider moving this to /packager/react-packager/src/node-haste/lib/getAssetDataFromName.js  so that if the function is used elsewhere the change is also effective.